libretro-rs/src/core_macro.rs

```diff
@@ -111,12 +111,12 @@ macro_rules! libretro_core {
       }
 
       #[no_mangle]
-      extern "C" fn retro_cheat_set(index: libc::c_uint, enabled: bool, code: *const libc::c_char) {
+      unsafe extern "C" fn retro_cheat_set(index: libc::c_uint, enabled: bool, code: *const libc::c_char) {
         instance_mut(|instance| instance.on_cheat_set(index, enabled, code))
       }
 
       #[no_mangle]
-      extern "C" fn retro_load_game(game: *const retro_game_info) -> bool {
+      unsafe extern "C" fn retro_load_game(game: *const retro_game_info) -> bool {
```

Comment:
Should these all be `unsafe` along with `instance_mut` and `instance_ref`? The latter two can't guarantee safety, so they might as well 🤷🏼‍♂️

Reply:
Yeah. I had that made that change in my api-2.0 branch, but I wanted to stick to the bare minimum changes to make Clippy happy on this one. The environment API is going to change after these two PRs get merged anyways; the callback-specific environments are going to break backwards compatibility and I'm going to add a `RetroEnvironmentData` marker trait to add a bit of additional safety rather than accepting literally any type.

Reply:
I may or may not have a super secret environment branch I've been working on 😅 Let's compare notes once I've got that done and see which parts we want to keep?

Reply:
Sounds good. I'm still experimenting in my api-2.0 branch while I break all the changes into PRs and get them merged.